### PR TITLE
Connected dApps: make right side of page-nav accomodate whatever we need

### LINKED ISSUE #19867: ⚓ Update page-top Quo component to accommodate connected dapps anchor
# Feature Issue

### User Story
As a developer,
I want to show a dapps link next to account menu on the account details page.

### Description
The current Quo component `page-nav` doesn't allow for adding actions next to the account switcher component. Account switcher is oddly named, it's actually the account menu.

But the new design needs us to show a link to connected dapps next to the account menu:

| Current | Required |
|--------|--------|
| image | image |
| Notice that the account switcher does not have any option for additional actions | Design needs the dapp action |

*Summary*: Upgrade the `page-nav` component and add an option to show an action.

### Acceptance Criteria
The page nav component should be update to show an action button.

### Notes
There are two approaches to this idea:
1. Update the `wallet-networks` type page nav to always show an action or
2. Create a new type `wallet-networks-with-action`

I'm inclined to take option 1 unless anyone disagrees. The reason is that the wallet-networks page nav will *always* have a link to the dapps. I can also make it configurable, ie the `icon` and the `on-press` handler can be passed as a prop.

### PR DESCRIPTION
**Pull request by alwx**
fixes https://github.com/status-im/status-mobile/issues/19867

### Summary

Updates `page-nav` to make it possible to accomodate different types of content on the right side of it — in this paticular case it means mixing `account-switcher` and regular buttons but the possible options are endless.

Also adds a button to open connected dapps screen + opens a very basic screen with just "Connected dApps" title. The rest will be handled in this issue: https://github.com/status-im/status-mobile/issues/19839
(all that is hidden behind the `wallet-connect` feature flag)

Documentation change PR (review please): https://github.com/status-im/status.im/pull/xxx

#### Platforms

- Android
- iOS

##### Functional

- wallet / transactions
- dapps / app browsing

status: ready

### CODE REVIEW

File: src/quo/components/navigation/page_nav/view.cljs [hunk 1]
```diff
@@ -44,48 +44,55 @@
...
+         :type                (or type :default)
+         :customization-color customization-color}]])))
+
+(defmethod add-button :default
+  [{:keys [background behind-overlay? button-props]}]
+  (let [{:keys [label icon-name]} button-props]
+    [button/button
+     (assoc button-props
+            :type       (button-type background)
+            :icon-only? icon-name
```
Comment (shivekkhurana) [inline src/quo/components/navigation/page_nav/view.cljs]: We use the `icon-name` for `icon-only?` prop which expects a boolean value. 

I suggest wrapping : `(keyword? icon-name)` or `(boolean icon-name)`. Both are fine.

File: src/status_im/contexts/wallet/connected_dapps/view.cljs [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+     :i/close]]
+   [quo/text
+    {:size   :heading-1
+     :weight :semi-bold
+     :style  (style/header-text (when subtitle true))}
+    title]])
+
+(defn view
+  []
+  [rn/view {:style {:flex 1}}
```
Comment (shivekkhurana) [inline src/status_im/contexts/wallet/connected_dapps/view.cljs:28]: Styles should be in style ns.

File: src/status_im/contexts/wallet/common/account_switcher/view.cljs [hunk 3]
```diff
@@ -18,21 +19,26 @@
...
       :accessibility-label accessibility-label
       :networks            networks
       :networks-on-press   #(rf/dispatch [:show-bottom-sheet {:content network-filter/view}])
-      :right-side          :account-switcher
-      :account-switcher    {:customization-color color
-                            :on-press            #(rf/dispatch [:show-bottom-sheet
-                                                                (get-bottom-sheet-args switcher-type)])
-                            :emoji               emoji
-                            :type                (when watch-only? :watch-only)}}]))
+      :right-side          [(when (ff/enabled? ::ff/wallet.wallet-connect)
```
Comment (shivekkhurana) [inline src/status_im/contexts/wallet/common/account_switcher/view.cljs]: The dapps button should not be visible for watch only account.

**Review by shivekkhurana — CHANGES_REQUESTED** [review]

**Comment by shivekkhurana** [comment]
Thank you for the feature and clever approach. This method is better than mine in #19929. 

I'm gonna close my PR and chill

Chilling...

File: src/status_im/contexts/wallet/connected_dapps/view.cljs (hunk 2 above)
Comment (clauxx) [reply src/status_im/contexts/wallet/connected_dapps/view.cljs:28]: I think there's a point in the guidelines about one-liner styles being allowed

**Review by shivekkhurana — APPROVED** [review]

**Comment by alwx** [comment]
To whoever will be testing this: need to make sure that `page-nav` doesn't break elsewhere, nothing else really needs to be checked because we're hiding this feature behind the feature flag now.

**Review by clauxx — APPROVED** [review]

**Comment by churik** [comment]
@alwx 
 thank you for detailed description!
 One question - do you know where `page-nav` is used? it is a part of basic navigation?

**Comment by qoqobolo** [comment]
Thanks for your work @alwx! I couldn't find any regression, PR can be merged.

**Pull request merged**